api.planx.uk/gis/digitalLand.ts

```diff
@@ -192,24 +196,24 @@ async function go(localAuthority, geom, extras) {
 
   // --- METADATA ---
   // additionally fetch metadata from Digital Land's "dataset" endpoint for extra context
-  let metadata = {};
+  const metadata: Record<string, Metadata> = {};
   const urls = activeDatasets.map((dataset) => `https://www.planning.data.gov.uk/dataset/${dataset}.json`);
   await Promise.all(urls.map(url => 
     fetch(url)
-      .then(response => response.json())
-      .catch(error => console.log(error))
+      .then((response: { json: () => any; }) => response.json())
+      .catch((error: any) => console.log(error))
   )).then((responses) => {
     responses.forEach((response) => {
       // get the planx variable that corresponds to this 'dataset', should never be null because we only requested known datasets
-      const key = Object.keys(baseSchema).find((key) => baseSchema[key]["digital-land-datasets"].includes(response.dataset));
-      metadata[key] = response;
+      const key = Object.keys(baseSchema).find((key) => baseSchema[key]["digital-land-datasets"]?.includes(response.dataset));
+      if (key) metadata[key] = response;
     });
   }).catch(error => console.log(error));
 
-  return { url: url, constraints: formattedResult, metadata: metadata };
+  return { url: url, constraints: formattedResult, metadata: metadata } as GISResponse;
```

Comment:
nit: We can avoid casting here by setting the return type of `go()` to be `Promise<GISResponse>`

Reply:
of course, will update :+1: